kube-runtime/src/controller/mod.rs

```diff
@@ -49,12 +49,44 @@ pub enum Error<ReconcilerErr: std::error::Error + 'static, QueueErr: std::error:
 
 /// Results of the reconciliation attempt
 #[derive(Debug, Clone)]
-pub struct ReconcilerAction {
+pub struct Action {
     /// Whether (and when) to next trigger the reconciliation if no external watch triggers hit
     ///
     /// For example, use this to query external systems for updates, expire time-limited resources, or
     /// (in your `error_policy`) retry after errors.
-    pub requeue_after: Option<Duration>,
+    requeue_after: Option<Duration>,
+}
+
+impl Action {
+    /// Action to to the reconciliation at this time even if no external watch triggers hit
+    ///
+    /// This is the best-practice action that ensures eventual consistency of your controller
+    /// even in the case of missed changes (which can happen).
+    ///
+    /// Watch events are not normally missed, so running this once per hour (`Default`) as a fallback is reasonable.
+    pub fn requeue(duration: Duration) -> Self {
+        Self {
+            requeue_after: Some(duration),
+        }
+    }
+
+    /// Do nothing until a change is detected
+    ///
+    /// This stops the controller periodically reconciling this object until a relevant watch event
+    /// was **detected**.
+    ///
+    /// **Warning**: If you have watch desyncs, it is possible to miss changes entirely.
+    /// It is therefore not recommended to disable requeuing this way, unless you have
+    /// frequent changes to the underlying object, or some other hook to retain eventual consistency.
+    pub fn await_change() -> Self {
+        Self { requeue_after: None }
+    }
+}
+
+impl Default for Action {
```

Comment:
Why the `Default` instance? Why an hour in particular?

Reply:
It's not necessary, but it felt like a nice number to point to. Desyncs are not particularly frequent these days. Have not actually seen one in ages in my controllers, so a pretty infrequent periodic might be a sensible value.

But happy to drop this, I felt it was nice to have during docs writing, but it's hard to nail down an accurate value. Kind of depends on app characteristics.

Reply:
have removed it. can't find justifications for a default in other sources.